posts/2021-03-25-Rust-1.51.0.md

```diff
@@ -0,0 +1,211 @@
+---
+layout: post
+title: "Announcing Rust 1.51.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.51.0. Rust is a
+programming language that is empowering everyone to build reliable and
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.51.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.51.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1510-2021-03-25
+
+## What's in 1.51.0 stable
+This release represents one of the largest additions to the Rust language and cargo in quite a while. With the stabilization of const generics, and the new feature resolver, along with more minor additions such as the `split-debuginfo` option, and the `addr_of!` API. Let's dive right into it!
+
+
+### Const Generics MVP
+Before this release, Rust allowed you to haves your types be parameterized over lifetimes or types. For example if we wanted to have a `struct` that is generic over an array of 32 items, we'd write the following:
+
+```rust
+struct FixedArray<T> {
+              // ^^^ Type generic definition
+    list: [T; 32]
+        // ^ Where we're using it.
+}
+```
+
+If we then use `FixedArray<u8>`, the compiler will make a monomorphic version of `FixedArray` that looks like:
+
+```rust
+struct FixedArray<u8> {
+    list: [u8; 32]
+}
+```
+
+This is a powerful feature that allows you to write reusable code with no runtime overhead. Enabling to have types that fit a wide variety of use-cases and types. However, until this release it hasn't been possible to easily be generic over the *values* of those types. This was most notable in arrays where they include their length in their type definition (`[T; N]`). Now with 1.51.0 you can write code that is generic over the values of any integer, `bool`, or `char` type! (Using `struct` or `enum` values is still unstable.)
+
+This change now lets us have our own array struct that's generic over its type *and* its length. Let's look at an example definition, and how it can be used.
+
+```rust
+struct Array<T, const LENGTH: usize> {
+    //          ^^^^^^^^^^^^^^^^^^^ Const generic definition.
+    list: [T; LENGTH]
+    //        ^^^^^^ We use it here.
+}
+```
+
+Now if we then used `Array<u8, 32>`, the compiler will make a monomorphic version of `Array` that looks like:
+
+```rust
+struct Array<u8, 32> {
+    list: [u8; 32]
+}
+```
+
+Const generics also allows to write our methods in a new way. Let's see what a `fn last(&self) -> Option<&T>` implementation could look like:
```

Comment:
Hm, can you say more about what this is intended to reference? I guess this example feels interesting but I'm not sure what it's trying to highlight about a 'new way' -- it was always possible to use constants in methods, just now we're generic, but it doesn't seem particularly different